### ISSUE TITLE
Addition of `List` type for fields

### ISSUE THREAD

**Issue by TheNoteTaker** [danielo515/obsidian-modal-form#313]
**Is your feature request related to a problem? Please describe.**

No.

**Describe the solution you'd like**

It's something that is not super high priority, but is a nice quality of life addition. The ability to have a `List` section which accepts multiple items and then returns an object whose value is an array of strings in post-processing for that key.

**Comment by danielo515** [comment]
Sorry, have you tried multi-select ? I think it does exactly what you
described

**Comment by TheNoteTaker** [comment]
Yes, I didn't realize you could type non-explicitly-defined results. I do have some other actual things I've found having used this plugin a bit more:

1. When you make a section conditional and then shift it up or down, the condition stays in place and does not shift with it. The new field which was shifted down (or up) will inherit that condition.
    - For example, `Foo` is in **Position 1** and `Bar` is in **Position 2**. `Bar` depends on `Foo`. When I shift `Bar` up to **Position 1**, and `Foo` naturally goes to **Position 2**, the conditional will still be at **Position 2** and make `Foo` now depend on `Bar`.
2. I have a setup with a `category` and a `subcategory`. 
    - `subcategory` depends on `category`
    - I then have a few fields which depend on `subcategory`. 
    - When I change `category`, `subcategory` disappears. The fields which depended on `subcategory` do not though because that value was set (even if for only a moment). Could you also clear out `subcategory` value when it's parent changes? This allows for chained conditionals.

**Comment by Xelbayria** [comment]
I was looking for a similar like this. so I can provide my suggestion. 

Multi-selection or select doesn't meet my fits. 

the form that don't have field for "list" in frontmatter so it would look like this: 

```
myLIst: 
  - sentence1
  - sentence2
```

in form, it would show one text to input sentence1. but if i want to input sentence2, so i'll "add" another text to input sentence2. not sure if that's possible to implement.

also to have 2 options for List field: an array or list. 

by array, i believe is `Mylist: [sentence1, sentence2]`  and the list is above with bullet or `-`, the minus. 

I tried to use textarea so i can input like this: 
```
- sentence1
  - sentence2
```
not exactly efficient. So here i am providing my suggestion.
